src/test/java/org/opensearch/geospatial/search/aggregations/bucket/geogrid/GeoHexGridParserTests.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.search.aggregations.bucket.geogrid;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.opensearch.geospatial.GeospatialTestHelper.randomHexGridPrecision;
+import static org.opensearch.geospatial.search.aggregations.bucket.geogrid.GeoHexGridAggregationBuilder.NAME;
+import static org.opensearch.geospatial.search.aggregations.bucket.geogrid.GeoHexGridAggregationBuilder.PARSER;
+
+import java.util.Locale;
+
+import org.hamcrest.MatcherAssert;
+import org.opensearch.common.xcontent.XContentParseException;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.geo.GeometryTestUtils;
+import org.opensearch.geometry.Rectangle;
+import org.opensearch.geospatial.h3.H3;
+import org.opensearch.test.OpenSearchTestCase;
+
+public class GeoHexGridParserTests extends OpenSearchTestCase {
+
+    public void testParseValidFromInts() throws Exception {
+        XContentParser stParser = createParser(
+            JsonXContent.jsonXContent,
+            String.format(
+                Locale.getDefault(),
+                "{\"field\":\"my_loc\", \"precision\":%d, \"size\": 500, \"shard_size\": 550}",
+                randomHexGridPrecision()
+            )
+        );
+        XContentParser.Token token = stParser.nextToken();
+        assertSame(XContentParser.Token.START_OBJECT, token);
+        // can create a factory
+        assertNotNull(PARSER.parse(stParser, NAME));
+    }
+
+    public void testParseValidFromStrings() throws Exception {
+        XContentParser stParser = createParser(
+            JsonXContent.jsonXContent,
+            String.format(
+                Locale.getDefault(),
+                "{\"field\":\"my_loc\", \"precision\":\"%d\", \"size\": \"500\", \"shard_size\": \"550\"}",
```

Comment:
looks like it's the same string we're using in multiple places, WDYT of making it a constant?

Reply:
I refactored to a method in some places.